pytissueoptics/scene/tests/logger/testLogger.py

```diff
@@ -1,46 +1,106 @@
+        self.assertEqual(0, len(logger.getDataPoints()))
+        self.assertEqual(0, len(logger.getSegments()))
 
     def testWhenLogNewPoint_shouldAddPointToTheLoggedPoints(self):
         logger = Logger()
-        logger.logPoint(Vector(0, 0, 0))
-        logger.logPoint(Vector(1, 0, 0))
+        logger.logPoint(Vector(0, 0, 0), self.INTERACTION_KEY)
+        logger.logPoint(Vector(1, 0, 0), self.INTERACTION_KEY)
         newPoint = Vector(2, 0, 0)
 
-        logger.logPoint(newPoint)
+        logger.logPoint(newPoint, self.INTERACTION_KEY)
 
-        self.assertEqual(3, len(logger.points))
-        self.assertEqual(newPoint, logger.points[-1])
+        self.assertEqual(3, len(logger.getPoints()))
+        self.assertEqual(newPoint, logger.getPoints()[-1])
 
     def testWhenLogNewDataPoint_shouldAddDataPointToTheLoggedDataPoints(self):
         logger = Logger()
-        logger.logDataPoint(-10.5, Vector(0, 0, 0))
-        logger.logDataPoint(20.5, Vector(1, 0, 0))
+        logger.logDataPoint(-10.5, Vector(0, 0, 0), self.INTERACTION_KEY)
+        logger.logDataPoint(20.5, Vector(1, 0, 0), self.INTERACTION_KEY)
 
-        logger.logDataPoint(10, Vector(2, 0, 0))
+        logger.logDataPoint(10, Vector(2, 0, 0), self.INTERACTION_KEY)
 
-        self.assertEqual(3, len(logger.dataPoints))
-        self.assertEqual(10, logger.dataPoints[-1].value)
-        self.assertEqual(Vector(2, 0, 0), logger.dataPoints[-1].position)
+        self.assertEqual(3, len(logger.getDataPoints()))
+        self.assertEqual(10, logger.getDataPoints()[-1].value)
+        self.assertEqual(Vector(2, 0, 0), logger.getDataPoints()[-1].position)
 
     def testWhenLogNewSegment_shouldAddSegmentToTheLoggedSegments(self):
         logger = Logger()
-        logger.logSegment(Vector(0, 0, 0), Vector(0, 0, 1))
-        logger.logSegment(Vector(0, 0, 0), Vector(0, 1, 0))
+        logger.logSegment(Vector(0, 0, 0), Vector(0, 0, 1), self.INTERACTION_KEY)
+        logger.logSegment(Vector(0, 0, 0), Vector(0, 1, 0), self.INTERACTION_KEY)
+
+        logger.logSegment(Vector(0, 0, 0), Vector(1, 0, 0), self.INTERACTION_KEY)
+
+        self.assertEqual(3, len(logger.getSegments()))
+        self.assertEqual(Vector(0, 0, 0), logger.getSegments()[-1].start)
+        self.assertEqual(Vector(1, 0, 0), logger.getSegments()[-1].end)
+
+    def testWhenGetDataWithKey_shouldReturnDataStoredForThisKey(self):
+        sameKey = InteractionKey(self.SOLID_LABEL, self.SURFACE_LABEL)
+        anotherKey = InteractionKey(self.SOLID_LABEL, "another surface")
+        anotherKeyWithoutSurface = InteractionKey(self.SOLID_LABEL)
+
+        logger = Logger()
+        logger.logPoint(Vector(0, 0, 0), self.INTERACTION_KEY)
+        logger.logPoint(Vector(1, 0, 0), sameKey)
+        logger.logPoint(Vector(2, 0, 0), anotherKey)
+        logger.logPoint(Vector(3, 0, 0), anotherKeyWithoutSurface)
+
+        self.assertEqual(2, len(logger.getPoints(self.INTERACTION_KEY)))
```

Comment:
would add `self.assertEqual(1, len(logger.getPoints(anotherKeyWithoutSurface)))`
to test for the None surface label ?